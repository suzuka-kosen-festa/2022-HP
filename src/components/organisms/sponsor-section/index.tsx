import type { FC } from "react";
import tw from "twin.macro";
import { SponsorSectionProperties } from "../../../models";
import { Heading } from "../../atoms/heading";
import { Text } from "../../atoms/text";

const SponsorSectionContainar = tw.section`space-y-[calc(200vw / 63)]`;

const SponsorSection: FC<SponsorSectionProperties> = ({ children, sponsors, title, ...rest }) => (
  <SponsorSectionContainar aria-label="協賛企業" {...rest}>
    <Heading>{title}</Heading>
    <Text css={tw`whitespace-pre-wrap`}>{children}</Text>
    <ul>
      {sponsors.map(item => (
        <Text as="li" css={tw`list-disc list-inside`} key={item}>
          {item}
        </Text>
      ))}
    </ul>
  </SponsorSectionContainar>
);

export { SponsorSection };
