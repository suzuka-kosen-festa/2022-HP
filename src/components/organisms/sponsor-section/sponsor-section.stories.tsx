import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { SponsorSection } from ".";

type T = typeof SponsorSection;
type Story = ComponentStoryObj<T>;

const data = {
  children: "今年の高専祭は3年ぶりの開催。\n以下の企業様が鈴鹿高専祭を応援してくれています！",
  sponsors: "・株式会社xx\n・...",
  title: "協賛企業",
};

export default {
  args: { children: data.children, sponsors: data.sponsors, title: data.title },
  component: SponsorSection,
} as ComponentMeta<T>;

export const Default: Story = {};
