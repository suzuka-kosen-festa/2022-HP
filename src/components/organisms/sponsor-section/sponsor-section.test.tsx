import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./sponsor-section.stories";

const { Default } = composeStories(stories);

const options = {
  name: "協賛企業",
};

describe("(components) organisms/sponsor-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("region", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
