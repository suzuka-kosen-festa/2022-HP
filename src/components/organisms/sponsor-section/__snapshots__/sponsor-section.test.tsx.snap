// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components) organisms/sponsor-section take snap shot 1`] = `
<div>
  <section
    aria-label="協賛企業"
    class="twin-1utt4d5-SponsorSectionContainar e1he0e0u0"
  >
    <h1
      class="twin-f7kb4u-Heading ebaq0s80"
    >
      協賛企業
    </h1>
    <p
      class="twin-zuqn7z-Text-SponsorSection epv4mo00"
    >
      今年の高専祭は3年ぶりの開催。
以下の企業様が鈴鹿高専祭を応援してくれています！
    </p>
    <ul
      class="twin-4njzar-SponsorSection"
    >
      <li
        class="twin-y76ykx-SponsorSection"
      >
        株式会社xx
      </li>
      <li
        class="twin-y76ykx-SponsorSection"
      >
        ...
      </li>
    </ul>
  </section>
</div>
`;
